jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -93,11 +76,102 @@ def on_limits_change(self, *args):
                     viewer.state.y_max = self.viewer.state.y_max
 
 
+class _BaseSelectZoom(BqplotSelectionTool):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.reset()
+
+    def _new_interact(self):
+        raise NotImplementedError(f"_new_interact not implemented by {self.__class__.__name__}")
+
+    def reset(self):
+        if hasattr(self, 'interact'):
+            self.interact.close()
+        self.interact = self._new_interact()
+        self.interact.observe(self.on_brushing, "brushing")
+
+    def on_brushing(self, msg):
+        if self.interact.brushing:
+            # start drawing a box, don't need to do anything
+            return
+
+        with delay_callback(self.viewer.state, 'x_min', 'x_max', 'y_min', 'y_max'):
+            # implement on_update_zoom to act on the applicable interact Selector
+            self.on_update_zoom()
+
+        # reset destroys the current interact and replaces with a new one to avoid having
+        # the draggable selection
+        self.reset()
+        # activate automatically activates the new interact so that another zoom is possible
+        # until deselecting the zoom tool
+        self.activate()
+
+
+@viewer_tool
+class BlinkOnce(Tool):
+    icon = os.path.join(ICON_DIR, 'blink.svg')
+    tool_id = 'jdaviz:blinkonce'
+    action_text = 'Go to next image'
+    tool_tip = ('Click on this button to display the next image, '
+                'or you can also press the "b" key anytime')
+
+    def activate(self):
+        self.viewer.blink_once()
+
+
+@viewer_tool
+class BoxZoom(_BaseSelectZoom):
+    icon = os.path.join(ICON_DIR, 'zoom_box.svg')
+    tool_id = 'jdaviz:boxzoom'
+    action_text = 'Box zoom'
+    tool_tip = 'Zoom to a drawn rectangle'
+
+    def _new_interact(self):
+        return BrushSelector(x_scale=self.viewer.scale_x,
+                             y_scale=self.viewer.scale_y,
+                             color=INTERACT_COLOR)
+
+    def on_update_zoom(self):
+        self.viewer.state.x_min, self.viewer.state.x_max = self.interact.selected_x
+        self.viewer.state.y_min, self.viewer.state.y_max = self.interact.selected_y
+
+
+@viewer_tool
+class XRangeZoom(_BaseSelectZoom):
```

Comment:
I don't see how this is useful in Imviz. Can you please clarify?

Reply:
this is used in the 2d spectrum viewer in mosviz (was just as easy to create it now while making sure the flexibility exists)

Reply:
I am just confused why it is hidden inside the `imviz` directory.

Reply:
ok, let me try to move all that are not imviz-specific into some generic location higher up (there does not yet seem to be a place for custom general tools, so I guess its time to make one).

Reply:
`jdaviz/core` or `jdaviz/configs/default`? 🤷

Reply:
done (moved all non-imviz to `jdaviz.core.tools` and left everything imviz-specific behind)